### PR TITLE
[main] autoscaling: make the min <= max validation fire when both fields are non-nil vs non-zero

### PR DESCRIPTION
**Pull request by thatmidwesterncoder**
Fixes a tiny outlier found by the UI testing - min <= max should fire always not just when both are non-zero, non-nil should be enough.

Added testcase for this as well. 

RFE: https://github.com/rancher/rancher/issues/49680

### CODE REVIEW

File: tests/v2prov/tests/autoscaler/autoscaler_test.go [hunk 1]
```diff
@@ -325,7 +325,33 @@ func Test_General_RKEMachinePool_Autoscaling_Field_Validation(t *testing.T) {
 					},
 				}
 			},
-			expectedError: "AutoscalingMinSize must be less than or equal to AutoscalingMaxSize when both are non-zero",
+			expectedError: "AutoscalingMinSize must be less than or equal to AutoscalingMaxSize when both are non-nil",
+			shouldSucceed: false,
+		},
+		{
+			name: "Invalid - EtcdRole with AutoscalingMinSize greater than AutoscalingMaxSize and Max set to zero",
```
Comment (jferrazbr) [inline tests/v2prov/tests/autoscaler/autoscaler_test.go:333]: Do you think its worth adding tests where only one autoscaling field is set (min present/max nil and max present/min nil) to assert the "both present" CEL semantics?

Comment (thatmidwesterncoder) [reply tests/v2prov/tests/autoscaler/autoscaler_test.go:333]: Yes - I added another CEL validation for that as well as tests surrounding it!

Also was able to figure out why the tests would flake sometimes - increasing the retry timeout seems to have fixed that.
